### PR TITLE
Avoid unnecessary ReferenceIdent & ColumnIdent allocations

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/AbstractInsertAnalyzer.java [hunk 1]
```diff
@@ -120,46 +119,34 @@ private boolean checkReferencesForGeneratedColumn(ColumnIdent columnIdent, Abstr
      *
      * the created column reference is returned
      */
-    protected Reference addColumn(String column, AbstractInsertAnalyzedStatement context, int i) {
-        assert context.tableInfo() != null;
-        return addColumn(new ReferenceIdent(context.tableInfo().ident(), column), context, i);
-    }
-
-    /**
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/AbstractInsertAnalyzer.java]: Why did you remove the comment?

**Comment by matriv** [comment]
Igtm otherwise

File: sql/src/main/java/io/crate/analyze/AbstractInsertAnalyzer.java (hunk 1 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/AbstractInsertAnalyzer.java]: I removed the whole function-overload. Now there is only one which contains the same comment

Comment (matriv) [inline sql/src/main/java/io/crate/analyze/AbstractInsertAnalyzer.java]: ok!
